week6/hamxxn/mission/src/pages/Stores/Stores.jsx

```diff
@@ -0,0 +1,60 @@
+import { useNavigate, useSearchParams } from "react-router-dom";
+import OrderBar from "../../components/OrderBar/OrderBar";
+import StatusBar from "../../components/StatusBar/StatusBar";
+import StoreDetail from "../../components/StoreDetail/StoreDetail";
+import { CategoryTitle, StoreDetailWrapper } from "./Store.styles";
+import { useEffect, useState } from "react";
+
+import { getStores } from "../../apis/stores";
+
+const addRankToStores = (stores) => {
+  const sortedStores = [...stores].sort((a, b) => b.rate - a.rate);
+
+  // 상위 3개에 rank 속성 추가
+  return sortedStores.map((store, index) => {
+    if (index < 3) {
+      return { ...store, rank: index + 1 };
+    }
+    return store;
+  });
+};
+const Stores = () => {
+  const [searchParams] = useSearchParams();
+  const category = searchParams.get("category");
+  const [stores, setStores] = useState([]);
+  //stores 정보 가져오기
+  useEffect(() => {
+    getStores().then((value) => setStores(value));
+    const sortStores = addRankToStores(stores);
+    setStores(sortStores);
```

Comment:
이렇게 되면 getStores에서 받아와서 setStores로 상태를 변경한 이후 또 다시 정려해서 setStores로 상태를 한 번 더 변경해서, 두 번 렌더링 되기도 하고 비동기적 특성 때문에 sort가 안 될 수도 있을 것 같아요

```suggestion
  useEffect(() => {
    getStores().then((value) => setStores(addRankToStores(value)));
```

이렇게 해서 한 번에 진행하는 것은 어떨까요?